src/CompareClasses.php

```diff
@@ -40,16 +38,16 @@ public function __invoke(
         ClassReflector $pastSourcesWithDependencies,
         ClassReflector $newSourcesWithDependencies
     ): Changes {
-        $definedApiClassNames = array_map(
-            static function (ReflectionClass $class): string {
-                return $class->getName();
-            },
-            array_filter(
+        $definedApiClassNames = Dict\map(
```

Comment:
Is there an advantage in using `Dict\map` over `array_map()` here? Here, the structure is not an iterable, so it would be better to rely on engine core concepts

Reply:
Same goes for `array_filter()`

Reply:
https://3v4l.org/KOgP7

Reply:
consistency ( in errors especially ) was the main reason for these functions, but at some point i forgot why i added them and made them "aliases" for the sake of performance ( see https://github.com/azjezz/psl/pull/169/files ), i think the middle ground is to wrap the closures again before passing them to builtin functions ( will release this as a bug fix in PSL ), see https://3v4l.org/TPJYP

Reply:
Wait what, `array_*` functions run with non-strict types all the times? O_O

Reply:
you have no idea about the amount of inconsistency i figured out while working on PSL.

Reply:
Cool, let's keep the PSL version then :+1:

Reply:
i will release a bug fix today in PSL ( https://github.com/azjezz/psl/issues/194 ), then i will finish this one :)